### PR TITLE
  CCD-2492: ccd-data-store-api Validate the scope of transactions in CCD

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
CCD-2492: ccd-data-store-api Validate the scope of transactions in CCD
            [CCD-2492] (https://tools.hmcts.net/jira/browse/CCD-2492).

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java [hunk 1]
```diff
@@ -138,7 +134,6 @@ public UIDraftsController(
         return ResponseEntity.ok(new CaseViewResource(caseView));
     }
 
-    @Transactional
     @DeleteMapping(path = "/drafts/{did}",
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java:137]: Is this method covered in Transactional scope in server layer or on repository?

File: src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java [hunk 2]
```diff
@@ -86,7 +84,6 @@ public UIDraftsController(
             caseTypeId));
     }
 
-    @Transactional
     @PutMapping(
         path = "/case-types/{ctid}/drafts/{did}",
         headers = {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java:89]: Is this method covered in Transactional scope in server layer or on repository?

File: src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java [hunk 3]
```diff
@@ -56,7 +55,6 @@ public UIDraftsController(
         this.draftGateway = draftGateway;
     }
 
-    @Transactional
     @PostMapping(
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java:58]: Is this method covered in Transactional scope in server layer or on repository?

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseDataValidatorController.java [hunk 4]
```diff
@@ -38,7 +37,6 @@ public CaseDataValidatorController(
         this.midEventCallback = midEventCallback;
     }
 
-    @Transactional
     @PostMapping(
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseDataValidatorController.java:40]: do we need PostMapping here?

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseAssignedUserRolesController.java [hunk 5]
```diff
@@ -135,7 +133,6 @@ public CaseAssignedUserRolesController(ApplicationParams applicationParams,
         return ResponseEntity.status(HttpStatus.CREATED).body(new CaseAssignedUserRolesResponse(ADD_SUCCESS_MESSAGE));
     }
 
-    @Transactional
     @DeleteMapping(
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseAssignedUserRolesController.java:136]: better to have transactional on delete methods, not on controller thought

File: src/main/java/uk/gov/hmcts/ccd/endpoint/std/CallbackEndpoint.java [hunk 6]
```diff
@@ -32,7 +31,6 @@ public CallbackEndpoint(final PrintableDocumentListOperation printableDocumentLi
         this.printableDocumentListOperation = printableDocumentListOperation;
     }
 
-    @Transactional
     @RequestMapping(value = "/jurisdictions/{jid}/case-types/{ctid}/documents", method = RequestMethod.POST)
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/std/CallbackEndpoint.java:34]: do we need POST for this request

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
minor comments

File: src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java (hunk 2 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UIDraftsController.java:89]: Yes it covered the implementation of executeUpdate is @Transactional

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseDataValidatorController.java (hunk 4 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseDataValidatorController.java:40]: well is it doing a invokeMidEventCallback() operations ,  it is not in the scope of  the ticket  removing it . we can do a call evaluated together :) and maybe we can create a new ticket.

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseAssignedUserRolesController.java (hunk 5 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseAssignedUserRolesController.java:136]: The implementation is CaseAccessOperation. removeCaseUserRoles is @Transactional

File: src/main/java/uk/gov/hmcts/ccd/endpoint/std/CallbackEndpoint.java (hunk 6 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/endpoint/std/CallbackEndpoint.java:34]: mmmm yes we need evaluate it . not 100 % sure .

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Comment by radhikahollakv** [comment]
Reverting Previous Commit

**Pull request merged**